app/models/task.py

```diff
@@ -1,6 +1,32 @@
 from flask import current_app
 from app import db
 
-
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    task_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    description = db.Column(db.String)
+    completed_at = db.Column(db.DateTime, nullable=True)
+    goal_id = db.Column(db.Integer, db.ForeignKey('goal.goal_id'), nullable=True)
+    goal = db.relationship('Goal', back_populates='tasks')
+
+    def to_dict(self):
+        if not self.completed_at:
+            self.completed_at = None
+        if not self.completed_at:
```

Comment:
Great work creating the correct response dictionary for the different situations. Consider how you could DRY up your code by first determining whether `"is_complete"` should be `True` or `False`